src/group_theory/order_of_element.lean

```diff
@@ -607,10 +622,9 @@ variables {H : Type u} [fintype H] [add_group H]
 lemma exists_gpow_eq_one (a : α) : ∃ i ≠ 0, a ^ (i : ℤ) = 1 :=
 begin
   rcases exists_pow_eq_one a with ⟨w, hw1, hw2⟩,
-  use w,
-  split,
-  { exact_mod_cast ne_of_gt hw1 },
-  { exact_mod_cast hw2 }
+  rw is_periodic_pt_mul_iff_pow_eq_one at hw2,
+  refine ⟨w, by exact_mod_cast ne_of_gt hw1, _⟩,
+  exact_mod_cast hw2,
```

Comment:
The following is also a proof (but I cannot suggest it, due to changes in-between):
```
  rcases exists_pow_eq_one a with ⟨w, hw1, hw2⟩,
  exact ⟨w, int.coe_nat_ne_zero.mpr (ne_of_gt hw1), (is_periodic_pt_mul_iff_pow_eq_one _).mp hw2⟩,
```